lms/services/launch_verifier.py

```diff
@@ -50,44 +49,77 @@ def verify(self):
             raise self._exception
 
     def _verify(self):
-        try:
-            consumer_key = self._request.params["oauth_consumer_key"]
-        except KeyError as err:
-            raise NoConsumerKey() from err
+        is_valid, _request = self._oauth1_endpoint.validate_request(
+            uri=self._request.url,
+            http_method=self._request.method,
+            body=self._request.body,
+            headers=dict(self._request.headers),
+        )
+
+        if not is_valid:
+            raise LTIOAuthError()
+
+
+class OAuthRequestValidator(RequestValidator):
+    # Value lifted from `oauth2` which is an `oauth1` library as was used by
+    # PyLTI. The default in `oauthlib` is 600
+    timestamp_lifetime = 300  # In seconds, five minutes.
+
+    # Tell oauthlib we are chill about http for local testing
+    enforce_ssl = False
+
+    def __init__(self, db_session):
+        super().__init__()
+        self.db_session = db_session
+
+    def check_client_key(self, client_key):
+        """Check that the client key only contains safe characters."""
+
+        # Just a style check for chars and length etc... we don't care about
+        # the format, as long as our lookup later works
+        return True
+
+    def check_nonce(self, nonce):
+        """Check that the nonce only contains only safe characters."""
+
+        # Just a style check for chars and length etc... we don't care about
+        # the format, as long as our lookup later works
+        return True
+
+    def validate_timestamp_and_nonce(
+        # pylint: disable=too-many-arguments
+        # Not our design, we have to fit in with this API
+        self,
+        client_key,
+        timestamp,
+        nonce,
+        request,
+        request_token=None,
+        access_token=None,
+    ):
+        """Validate that the nonce has not been used before."""
+
+        # We don't appear to care about the nonce value
+        return True
+
+    def validate_client_key(self, client_key, request):
+        """Validate that supplied client key is registered and valid."""
+
+        # This is slightly incorrect, but we are just going to say the client
+        # key is fine! Even though this might not be true. This is because
+        # we would have to look up the DB to find out, and we _have_ to do
+        # that in the next step `get_client_secret()`
+
+        return True
+
+    def get_client_secret(self, client_key, request):
+        """Retrieve the client secret associated with the client key."""
 
         application_instance = models.ApplicationInstance.get_by_consumer_key(
-            self._request.db, consumer_key
+            self.db_session, client_key
         )
 
         if not application_instance:
             raise ConsumerKeyError()
```

Comment:
If we want any feedback about what's going wrong in this process we have to raise exceptions or `oauthlib` will eat all the feedback.